src/modals/feedback_modal.js

```diff
@@ -0,0 +1,66 @@
+import FeedbackModalView from '../views/feedback_modal.dot'
+import ExtendedString from "../libs/string"
+import Modal from "./modal"
+
+export default class FeedbackModal {
+  constructor(results) {
+    this.modal = new Modal(this, FeedbackModalView)
+
+    this.ExtendedString = ExtendedString
+
+    this.original_results = []
+    this.displayed_results = []
+    this.current_query = ''
+
+    this.fields = {
+      result_not_found: false,
+      result_not_found_details: '',
+      result_duplicate: false,
+      result_bad_ranking: false,
+      result_wrong_geolocation: false,
+      result_other: false,
+      result_other_details: '',
+    }
+
+    this.selected_result = {}
+
+    listen('open_feedback_modal', (results) => {
+      this.current_query = window.__searchInput.searchInputHandle.value
+      this.original_results = this.displayed_results = results || []
+      this.displayed_results.forEach((res, idx) => {
+        res.icon = IconManager.get({className : res.className, subClassName : res.subClassName , type : res.type})
```

Comment:
Please remove whitespace before ':' and before ','.

Reply:
Done.